independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Injection.java

```diff
@@ -40,13 +40,55 @@
         if (Kind.CLASS.equals(beanTarget.kind())) {
             List<Injection> injections = new ArrayList<>();
             forClassBean(beanTarget.asClass(), beanTarget.asClass(), beanDeployment, injections, transformer, false);
+
             Set<AnnotationTarget> injectConstructors = injections.stream().filter(Injection::isConstructor)
                     .map(Injection::getTarget).collect(Collectors.toSet());
             if (injectConstructors.size() > 1) {
                 throw new DefinitionException(
                         "Multiple @Inject constructors found on " + beanTarget.asClass().name() + ":\n"
                                 + injectConstructors.stream().map(Object::toString).collect(Collectors.joining("\n")));
             }
+
+            Set<MethodInfo> initializerMethods = injections.stream()
+                    .filter(it -> it.isMethod() && !it.isConstructor())
+                    .map(Injection::getTarget)
+                    .map(AnnotationTarget::asMethod)
+                    .collect(Collectors.toSet());
+            for (MethodInfo initializerMethod : initializerMethods) {
+                if (beanDeployment.hasAnnotation(initializerMethod, DotNames.PRODUCES)) {
+                    throw new DefinitionException("Initializer method must not be marked @Produces "
+                            + "(alternatively, producer method must not be marked @Inject): "
+                            + beanTarget.asClass() + "." + initializerMethod.name());
+                }
+
+                if (beanDeployment.getAnnotations(initializerMethod)
```

Comment:
Maybe we could add a util method similar to `io.quarkus.arc.processor.Annotations.getParameterAnnotations(BeanDeployment, MethodInfo, int)`?

Reply:
Actually I have to rewrite it a little, right now it only looks into Jandex data and not into the annotation store. Hold on a sec.

Reply:
Done now.